docs/guides/integration-javascript.md

```diff
@@ -0,0 +1,293 @@
+
+The UID2 SDK hashes the phone number before sending the encrypted hash to the UID2 service.
+
+### Configure for Hashed Phone Number
+
+Configure the UID2 SDK with a hashed phone number:
+
+```js
+await __uid2.setIdentityFromPhoneHash(
+    'eVvLS/Vg+YZ6+z3i0NOpSXYyQAfEXqCZ7BTpAjFUBUc=',
+    {
+        subscriptionId: subscriptionId,
+        serverPublicKey: publicKey,
+    }
+);
+```
+
+**The publisher is responsible for normalizing and hashing the phone number**. For details, see [Normalization and Encoding](../getting-started/gs-normalization-encoding.md).
+
+The UID2 SDK encrypts the hash before sending it to the UID2 service.
+
+## Token Storage and Refresh
+
+The generated UID2 token will be stored in local storage, under the key UID2-sdk-identity. The SDK will refresh the UID2 token periodically.
+
+## Retrieving the UID2 Advertising Token 
+
+With the setup above, when the UID2 SDK generates a UID2 token, the advertising token (TODO add link to glossary) will be ready to retrieve by calling the ```getAdvertisingToken```. The advertising token is the token publishers will pass to the bidstream. Before calling this function, be sure to call init() and wait until your callback handler has received an InitCompleted event.
+
+```js
+<script>
+  let advertisingToken = __uid2.getAdvertisingToken();
+</script>
+```
+
+See more info and typescript example on [getAdvertisingToken API reference](../sdks/client-side-identity#getadvertisingtoken-string)
+
+There is another call ```getAdvertisingTokenAsync``` which can return a ```Promise``` string for the latest advertising token.   
+
+See more info and typescript example on [getAdvertisingTokenAsync API reference](../sdks/client-side-identity#getadvertisingtokenasync-promise)
+
+```js
+<script>
+  __uid2.getAdvertisingTokenAsync()
+    .then(advertisingToken => { /* initiate targeted advertising */ })
+    .catch(err => { /* advertising token not available */ });
+</script>
+```
+
+## When to Pass DII to the UID2 SDK
+
+If possible, configure the UID2 SDK with the user's DII on each page load.
+
+When the UID2 SDK is configured, it checks for an existing UID2 token in the user's browser. If there is an existing token that was generated from the same DII, and the token is still valid or can be refreshed, the SDK uses or refreshes the existing token instead of generating a new token.
+
+If there is no existing token, or the token has expired and cannot be refreshed, the UID2 SDK cannot generate a new token without DII.
+
+As a result, the recommended approach is to call the ```setIdentityFromXXXX``` method with the user's DII on each page load.
+
+In some cases, the user's DII is not available on page load, and getting the DII has some associated cost. For example, an API call might be required to fetch the DII, or the user has to be prompted to provide the DII information.
+
+You can potentially avoid that cost by checking for an existing token that you can use or refresh - simply calling
```

Comment:
is this the correct suggestion @lionell-pack-ttd ?

Reply:
I think we should be providing examples based on the events - but we don't have a good event for this case yet. If `InitCompleted` contains no identity, there might still be an active request trying to load one.

`getAdvertisingToken` provides no useful information at all about this. The only viable option right now is `getAdvertisingTokenAsync`.

Reply:
i have given it a shot to provide examples and explanation. please have a read again and see if it makes sense.

Reply:
The explanation and example looks good - but see Gen's comment, it does look like just one snippet but the explanation says "snippets" plural.

Reply:
thanks i fixed it